src/lightning/fabric/strategies/deepspeed.py

```diff
@@ -350,10 +350,11 @@ def module_init_context(self, empty_init: Optional[bool] = None) -> ContextManag
             raise NotImplementedError(
                 f"`{empty_init=}` is not a valid choice with `DeepSpeedStrategy` when ZeRO stage 3 is enabled."
             )
+        module_sharded_ctx = self.module_sharded_context()
         stack = ExitStack()
         if not self.zero_stage_3:
             stack.enter_context(super().module_init_context(empty_init=empty_init))
```

Comment:
for completeness, it would probably also be better to apply it to this line right?

Reply:
I don't think this one matters because in the super() call, all the ctxmanagers are instantiated before any is entered